packages/veritone-widgets/src/widgets/FolderSelectionDialog/CollapsibleFolder.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { objectOf, func, any } from 'prop-types';
+import { connect } from 'react-redux';
+import * as folderSelectionModule from '../../redux/modules/folderSelectionDialog';
+import cx from 'classnames';
+import FolderIcon from '@material-ui/icons/Folder';
+import FolderList from './FolderList';
+import ExpandLess from '@material-ui/icons/ArrowDropDown';
+import ExpandMore from '@material-ui/icons/ArrowRight';
+import styles from './styles.scss';
+
+
+
+@connect(
+  (state) => ({
+    selectedFolder: folderSelectionModule.selectedFolder(state),
+  }),
+  {
+    selectFolder: folderSelectionModule.selectFolder,
+    getAllSubFolders: folderSelectionModule.getAllSubFolders
+  }
+)
+
+export default class CollapsibleFolder extends React.Component {
+  static propTypes = {
+    folder: objectOf(any),
+    selectFolder: func,
+    getAllSubFolders: func,
+    selectedFolder: objectOf(any)
+  };
+
+  state = { 
+    open: false,
+  }
+
+
+  handleClick = () => {
+    // open or close collapsible folder - refetch subfolders whenever opening but not closing and select the folder the clicked on
+    const { folder, selectFolder, getAllSubFolders } = this.props;
+    if(!this.state.open){
+      getAllSubFolders(folder);
+    }
+    this.setState((prevState) => ({
+      open : !prevState.open,
+    }));
+    selectFolder(folder);
+  }
+
+
+  render() {
+    // the collapsible folder has subfolders which are rendered when open by passing a listId into FolderList component
+    const { folder, selectedFolder } = this.props;
+    let listId  = folder.treeObjectId;
+    let selectedId = selectedFolder.treeObjectId;
+
+    return (
+        <li>
+          <div className={cx(styles.folder, listId === selectedId && styles.selected)} onClick={this.handleClick}>
+            {this.state.open? <ExpandLess style ={{fontSize: "30px"}}/> : <ExpandMore style ={{fontSize: "30px"}}  />}
+       
+            <FolderIcon className={styles.collapsibleFolderIcon}/>
+         
+            <div className={cx(styles.folderName, listId === selectedId  && styles.folderNameSelected)}>{folder.name}</div>
```

Comment:
For readability purpose, can we assign 'listId === selectedId' to variable and show it here?

Reply:
good suggestion